### DISCUSSION TITLE
Shared album, making sense of comments and likes/favorites (or how to collaboratively pull selects)

### DISCUSSION THREAD

**Discussion by curiousercreative** [immich-app/immich#5514] [Q&A]
Hello, new to Immich, thanks very much for this wonderful software and seemingly vibrant community.

This is my first install of Immich and obviously not exactly what it's built for. I want a shared photo collection (using a shared album currently) where multiple users can upload, comment, and like/favorite/rate the photos in the collection towards and ultimate goal of collaboratively selecting 12 photos (from hundreds) for a calendar. Immich has most of this covered, but after users have liked/favorited/commented on photos in the album, there's nothing beyond a simple activity log and thumbnail for each of these actions. Can anyone recommend a way (happy to receive hacking suggestions, db queries, codebase mods, etc) to accomplish any of the following?
1. show the count and/or who liked/commented on each photo in the main album view
1. hyperlink the liked/comment activity log entries such that I can click on the entry or thumbnail and get to the actual photo
1. filter or sort the album view based on has likes, has comments, # of likes, # of comments or something similar
1. a DB query that would get me original filenames and the like or comment count

Thanks in advance!

p.s. if you know of a purpose built alternative tool for this (similarly FOSS please!), let me know :)

**Comment by alextran1502** [comment]
Hello, what you are proposing is perhaps similar to Facebook's system if I understand correctly, right?

**Reply by curiousercreative** [reply]
I don't know, I haven't uploaded a photo album to Facebook in 15 years

**Comment by curiousercreative** [comment]
Here's a [quick gist](https://gist.github.com/curiousercreative/a75037c15e891000dac6823c28d3204e) I'll use to partially accomplish 3 and less so 1. I initially attempted to solve 1 above but the images are lazy loaded only what's in viewport. I haven't figured out how to accomplish 2 via the activity panel or via the thumbs my gist renders.

**Reply by lord-xeon** [reply]
@curiousercreative that gist was exactly what I was looking for - I need to do #3 to be able to select all the photos that had been liked.

[I had to modify it a bit](https://gist.github.com/lord-xeon/c839c0b4b6dd6336c621d2625e69e279) due to changes in the way Immich renders the dom, but it works on **version 1.114.0**.
